scripts/restore

```diff
@@ -1,99 +1,115 @@
-# RESTORE APT PREFERENCES AND SOURCES.LIST
+# RESTORE USER RIGHTS
 #=================================================
-ynh_restore_file "/etc/apt/sources.list.d/$app.list"
 
-ynh_restore_file "/etc/apt/preferences.d/00MinetestPinning"
+# Restore permissions on app files
+chown -R root: $final_path
+chown -R $app: /home/yunohost.app/$app
 
 #=================================================
-# INSTALL
+# SPECIFIC RESTORATION
 #=================================================
-if [ $(uname -m) == "armv7l" ]
-then
-    gpg --list-keys 7638D0442B90D010 > /dev/null 2>&1
-    if [ $? != 0 ]
-    then
-        gpg --keyserver pgpkeys.mit.edu --recv-key 7638D0442B90D010
-    fi
-    gpg --list-keys 8B48AD6246925553 > /dev/null 2>&1
-    if [ $? != 0 ]
-    then
-        gpg --keyserver pgpkeys.mit.edu --recv-key 8B48AD6246925553
-    fi
-    if [ $(apt-key finger | grep "7638 D044 2B90 D010" -c) == 0 ]
-    then
-        gpg -a --export 7638D0442B90D010 | apt-key add -
-    fi
-    if [ $(apt-key finger | grep "8B48 AD62 4692 5553" -c) == 0 ]
-    then
-        gpg -a --export 8B48AD6246925553 | apt-key add -
-    fi
-fi
-ynh_package_update
-ynh_install_app_dependencies minetest-server sqlite3
+# REINSTALL DEPENDENCIES
+#=================================================
+ynh_print_info --message="Reinstalling dependencies..."
+
+# Define and install dependencies
+ynh_install_app_dependencies $pkg_dependencies
 
 #=================================================
-# ENABLE SERVICE IN ADMIN PANEL
+# RESTORE SYSTEMD
 #=================================================
+ynh_print_info --message="Restoring the systemd configuration..."
 
-# Add service to Yunohost's monitoring
-yunohost service add minetest --log "/var/log/minetest/minetest.log"
+ynh_restore_file --origin_path="/etc/systemd/system/$app.service"
+systemctl enable $app.service
 
 #=================================================
-# STOP MINETEST
+# ADVERTISE SERVICE IN ADMIN PANEL
 #=================================================
-service minetest-server stop
+
+yunohost service add $app --log "/var/log/$app/$app.log"
+
 
 #=================================================
-# CONFIGURATION
+# RESTORE THE LOGROTATE CONFIGURATION
 #=================================================
-ynh_secure_remove "/etc/minetest/minetest.conf"
-ynh_restore_file "/etc/minetest/minetest.conf"
 
+ynh_restore_file --origin_path="/etc/logrotate.d/$app"
+
+#=================================================
+# GENERIC FINALIZATION
 #=================================================
-# DATA
+# RELOAD MINETEST
 #=================================================
-ynh_secure_remove "/var/games/minetest-server/"
-ynh_restore_file "/var/games/minetest-server/"
+ynh_print_info --message="Reloading minetest..."
 
-ynh_restore_file "/tmp/map.sqlite"
-echo ".restore /tmp/map.sqlite" | sqlite3 /var/games/minetest-server/.minetest/worlds/world/map.sqlite
+systemctl restart $app
```

Comment:
Remove this systemctl restart $app, and ynh_systemd_action, and replace it with:

`ynh_systemd_action --action=restart --line_match="Server for gameid="$app" listening on :::30000." --log_path="/var/log/$app/minetest.log"`

`sleep 1` shouldn't be needed after that